std/algorithm/searching.d

```diff
@@ -233,14 +233,25 @@ bool balancedParens(Range, E)(Range r, E lPar, E rPar,
 if (isInputRange!(Range) && is(typeof(r.front == lPar)))
 {
     size_t count;
-    for (; !r.empty; r.popFront())
+
+    static if (is(Unqual!(ElementEncodingType!Range) == Unqual!E) && isNarrowString!Range)
+    {
+        import std.utf : byCodeUnit;
+        auto rn = r.byCodeUnit;
+    }
+    else
     {
-        if (r.front == lPar)
+        alias rn = r;
+    }
+
+    for (; !rn.empty; rn.popFront())
```

Comment:
You could have used the opportunity to use a `foreach` here ;-)

Reply:
I disagreed with the rejection of `for` for DMD, so I'm not going to go through the trouble of removing it when it's already there.